src/transport/grpc/mod.rs

```diff
@@ -649,9 +649,11 @@ mod test {
 
     use super::*;
 
+    const ID: Uuid = uuid!("67e55044-10b1-426f-9247-bb680e5fe0c8");
+
     #[derive(Debug)]
     enum ServerReceivedRequest {
-        Attach(Node),
+        Attach((Uuid, Node)),
```

Comment:
Why did you add uuid here? Is not possible to read from `get_node_id_from_metadata`?